src/Fable.Transforms/Rust/Fable2Rust.fs

```diff
@@ -1129,10 +1149,25 @@ module Util =
         let callee = mkGenericPathExpr [cellTy;"from"] None
         mkCallExpr callee [value]
 
-    // For any ref counted types, clone unconditionally (should be used only when returning references)
+    // For any ref counted types, clone when passing over a boundary, binding, closing over, etc
     let transformExprCloneRef (com: IRustCompiler) ctx (e: Fable.Expr): Rust.Expr =
         let expr = com.TransformAsExpr (ctx, e)
-        if shouldBeRefCountWrapped com e.Type then
+        let isOnlyReference =
```

Comment:
I noticed you simplified this @ncave , but this seems to be introducing extra unnecessary clones when returning. Is this definitely right?

Reply:
I did (as a separate function, the other one was not used anymore). If return is a reference to Rc, it should be cloned, no?

Reply:
Its not quite that simple, hence why I think I called it maybe.

If the thing is being returned from a function, you can guarantee it is the only reference because it is not yet bound to anything, so you don't need to clone it. If it is also the only reference in scope as inferred from the context, you also don't need to clone it.

Reply:
hmm now im not sure myself lol. I wonder if all these calls could just be redirected onto transformLeaveContextByValue

Reply:
I mean it was broken, I added to an Array test to highlight the issue. A lambda was taking a `&Rc` param and trying to return it without .clone().

Reply:
Is it working for you now?

Reply:
Yes, it works. I understand it needs to be conditional to avoid extra clones when not needed.

Reply:
I see, no worries.

Ok yes I just tested, if all transformExprCloneRef are removed and replaced by transformLeaveContextByValue, it does the smarter checks to only clone if it really needs to.

Want me to put up a quick PR to fix this?

Reply:
Sure, but please also test with function (not lambda) returns.

Reply:
I just tested it, functions are fine.

Reply:
Yay. Something is screwy with nested let bindings though, they are still cloning. Looks like this was broken before our merge issues, so we must be missing a exemption match in the list. Have you got any idea what a block statement with a return looks like in the AST?

Here is an example:
```
              let tupledArg_1: Rc<(i32, i32)> =
                            {
                                let tupledArg: Rc<(i32, i32)> = x.clone();
                                destructureAndAdd1(tupledArg.0, tupledArg.1)
                            }.clone();
```

Reply:
Can you show the F# source for this? There is (there was at least) explicit cloning in let statements and function/lambda returns before your latest PR.